From f54bccb3918103b385b99b993e076ddf3a35e8ea Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sun, 13 Jan 2019 17:23:22 +0200
Subject: [PATCH] [SW-0]: add Wall and Werror to CFLAGS

This reveal set but not used variables issue that was fixed as well

Change-Id: I68d8af4b2ffebb279b448363f5d39fdc17ab6440
---

diff --git a/Makefile b/Makefile
index 291e074..e6410cb 100644
--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,7 @@
 CFLAGS += -I ../uboot-custom/arch/arm/include
 CFLAGS += -I ../uboot-custom/arch/arm/mach-habanalabs/include
 CFLAGS += -include ../uboot-custom/include/linux/kconfig.h
+CFLAGS += -Wall -Werror
 CFLAGS += -c
 ifeq ($(HL_DEV),goya)
 	CFLAGS += -DDEV_IS_GOYA
diff --git a/preboot.c b/preboot.c
index f6b2e61..55b44d0 100644
--- a/preboot.c
+++ b/preboot.c
@@ -302,7 +302,6 @@
 static void spi_special_cmd_csel(unsigned int cmd, unsigned char *val, size_t size)
 {
 	int i;
-	unsigned char tmp;
 
 	spi_claim_bus();
 
@@ -316,10 +315,10 @@
 	}
 
 	while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
-	tmp = readb(mmPSOC_SPI_DR0);
+	readb(mmPSOC_SPI_DR0);
 	for (i=0; i<size; i++) {
 		while (spi_fifo_is_empty(mmPSOC_SPI_RXFLR));
-		tmp = readb(mmPSOC_SPI_DR0);
+		readb(mmPSOC_SPI_DR0);
 	}
 
 	spi_release_bus();
